### DISCUSSION TITLE
Understanding the Use of @js.native in generated Scala.js code

### DISCUSSION THREAD

**Discussion by kimiakh** [ScalablyTyped/Converter#576] [Q&A]
I've been working with ScalablyTyped to create a façade; I've noticed that in some cases, the generated Scala.js code (from d.ts files) includes the @js.native annotation, while in others, it doesn't. I'm looking to better understand the criteria or conditions under which @js.native is added to the generated code and when it's not. So, what determines whether the generated Scala.js traits include @js.native or not?

**Comment by oyvindberg** [comment]
Hey there!

Yeah, this is unfortunately pretty convoluted. The logic is here https://github.com/ScalablyTyped/Converter/blob/master/importer-portable/src/main/scala/org/scalablytyped/converter/internal/importer/IsUserImplementable.scala . I've tried to comment it, but feel free to ask follow-up questions as you have them :)
